data/abi/abi_json.go

```diff
@@ -88,24 +123,19 @@ func (t Type) MarshalToJSON(value interface{}) ([]byte, error) {
 		}
 		return json.Marshal(byteValue)
 	case Address:
-		var addressValueInternal []byte
+		var addressBytes [addressByteSize]byte
 		switch valueCasted := value.(type) {
 		case []byte:
 			if len(valueCasted) != addressByteSize {
 				return nil, fmt.Errorf("address byte slice length not equal to 32 byte")
 			}
-			addressValueInternal = valueCasted
+			copy(addressBytes[:], valueCasted[:])
```

Comment:
@jasonpaulos For my edification - How did you identify the bug?

Reply:
Essentially, golang slice does not allow one to copy from a length-32 slice to a nil length slice, and I was wrongly assuming the flexibility.

Reply:
@ahangsu is correct about the bug location.

I was working on creating explicit `addressToString` and `addressFromString` functions so that they could be better tested without the `data/basics` package, when I discovered this bug.

Reply:
@jasonpaulos Got it - thanks for the context.  Resolving.